src/EditorFeatures/Test/EditAndContinue/CompileTimeSolutionProviderTests.cs

```diff
@@ -80,5 +81,78 @@ public async Task TryGetCompileTimeDocumentAsync(string kind)
 
             AssertEx.Equal(new[] { documentId, designTimeDocumentId }, actualDesignTimeDocumentIds);
         }
+
+        [Fact]
+        public async Task GeneratorOutputCachedBetweenAcrossCompileTimeSolutions()
+        {
+            var workspace = new TestWorkspace(composition: FeaturesTestCompositions.Features);
+            var projectId = ProjectId.CreateNewId();
+
+            var generatorInvocations = 0;
+
+            var generator = new PipelineCallbackGenerator(context =>
+            {
+                // We'll replicate a simple example of how the razor generator handles disabling here so the test
+                // functions similar to the real world
+                var isDisabled = context.AnalyzerConfigOptionsProvider.Select(
+                    (o, ct) => o.GlobalOptions.TryGetValue("build_property.SuppressRazorSourceGenerator", out var value) && bool.Parse(value));
+
+                var sources = context.AdditionalTextsProvider.Combine(isDisabled).Select((pair, ct) =>
+                {
+                    var (additionalText, isDisabledFlag) = pair;
+
+                    if (isDisabledFlag)
+                        return null;
+
+                    Interlocked.Increment(ref generatorInvocations);
+                    return "// " + additionalText.GetText(ct)!.ToString();
+                });
+
+                context.RegisterSourceOutput(sources, (context, s) =>
+                {
+                    if (s != null)
+                        context.AddSource("hint", SourceText.From(s));
+                });
+            });
+
+            var analyzerConfigId = DocumentId.CreateNewId(projectId);
+            var additionalDocumentId = DocumentId.CreateNewId(projectId);
+
+            var analyzerConfigText = "is_global = true\r\nbuild_property.SuppressRazorSourceGenerator = true";
+
+            workspace.SetCurrentSolution(s => s.
+                AddProject(ProjectInfo.Create(projectId, VersionStamp.Default, "proj", "proj", LanguageNames.CSharp)).
+                AddAnalyzerReference(projectId, new TestGeneratorReference(generator)).
+                AddAdditionalDocument(additionalDocumentId, "additional", SourceText.From(""), filePath: "additional.razor").
+                AddAnalyzerConfigDocument(analyzerConfigId, "config", SourceText.From(analyzerConfigText), filePath: "Z:\\RazorSourceGenerator.razorencconfig"),
+                WorkspaceChangeKind.SolutionAdded);
+
+            // Fetch a compilation first for the base solution; we're doing this because currently if we try to move the
+            // cached generator state to a snapshot that has no CompilationTracker at all, we won't update the state.
```

Comment:
This is one limitation with the approach here: the API to move the generator state from one to another is being implemented via a translation action. If you're calling it on a project that has no compilation at all, we throw away the translation action rather than creating a CompilationTracker for it. Although this can be addressed, I'm not sure the extra code is necessarily worth the additional complexity since I don't think we'll see that case happen in the real world.